### PR TITLE
update-insecure-requests CSP demo.

### PR DESCRIPTION
**Pull request by jeffposnick**
@PaulKinlan @addyosmani @mikewest

Here's a demo of using `<meta http-equiv="Content-Security-Policy" content="upgrade-insecure-requests">` to put the new CSP into effect, along with a simple image embed showing off the functionality.

You can view a live version at https://rawgit.com/jeffposnick/samples/csp-upgrade-insecure-requests/csp-upgrade-insecure-requests/index.html

### CODE REVIEW

File: csp-upgrade-insecure-requests/index.html [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    </p>
+
+    <p>
+      Like other Content Security Policies, the recommend approach is to enable it via a HTTP
+      response header, <code>Content-Security-Policy: upgrade-insecure-requests</code>. However,
+      if you do not have control over the underlying web server (as is the case in this demo), an
+      <a href="http://www.html5rocks.com/en/tutorials/security/content-security-policy/#the-meta-tag">alternative</a>
+      is to include the
+      <code>&lt;meta http-equiv="Content-Security-Policy" content="upgrade-insecure-requests"&gt;</code>
+      tag in your HTML, preferably in the <code>&lt;head&gt;</code>.
```
Comment (mikewest) [inline csp-upgrade-insecure-requests/index.html]: It has to be inside `<head>`, we'll ignore the policy if it appears in the body.

(Not sure you need to go into the detail, but you can inject it into the head via JavaScript; all that matters is that `<head>` is an ancestor of `<meta>`.)

File: csp-upgrade-insecure-requests/index.html [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    </p>
+
+    <div class="output">
+      <p>
+        The following image is loaded with an explicit <code>http:</code> URL,
+        <code>http://googlechrome.github.io/samples/images/touch/chrome-touch-icon-192x192.png</code>.
+        Because this page has <code>Content-Security-Policy: upgrade-insecure-requests</code>
+        active, the <code>http:</code> is treated as <code>https:</code>, and no mixed-content
+        warnings are displayed.
```
Comment (mikewest) [inline csp-upgrade-insecure-requests/index.html:94]: Perhaps you could force the page onto HTTPS via JavaScript to make sure this is a true statement? :) (e.g. `if (window.location.protocol != "https:") { window.location.protocol = "https:"; }`)

Comment (jeffposnick) [inline csp-upgrade-insecure-requests/index.html:94]: Makes sense. I put in the same redirect logic we're using for the service worker demos.

**Comment by addyosmani** [comment]
Now that Mike's suggested changes are in, sample LGTM. Thanks Jeff!

**Pull request merged**